e2e/machine_migration_test.go

```diff
@@ -110,6 +112,444 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 				verifyCAPIMachinePausedCondition(newCapiMachine, machinev1beta1.MachineAuthorityClusterAPI)
 			})
 		})
+
+		Context("with spec.authoritativeAPI: ClusterAPI, Prevent changes to non-authoritative Machines except from sync controller", func() {
+			var testMachineName = "machine-vap-83955"
+			var testMapiMachine *machinev1beta1.Machine
+			var testCapiMachine *clusterv1.Machine
+
+			BeforeAll(func() {
+				testMapiMachine = createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				DeferCleanup(func() {
+					By("Cleaning up machine resources")
+					cleanupMachineResources(
+						ctx,
+						cl,
+						[]*clusterv1.Machine{testCapiMachine},
+						[]*machinev1beta1.Machine{testMapiMachine},
+					)
+				})
+			})
+
+			It("should verify CAPI Machine gets created and becomes Running", func() {
+				verifyMachineRunning(cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+			})
+
+			It("should verify that the non-authoritative MAPI Machine has an authoritative CAPI Machine mirror", func() {
+				Eventually(func() error {
+					testCapiMachine, err = capiframework.GetMachine(cl, testMachineName, capiframework.CAPINamespace)
+					return err
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(Succeed(), "CAPI Machine should exist")
+			})
+
+			It("should prevent modification of InstanceType in non-authoritative MAPI Machine", func() {
+				By("Attempting to modify InstanceType in MAPI Machine")
+
+				// Get the latest version of the machine to avoid resourceVersion conflicts
+				currentMachine := &machinev1beta1.Machine{}
+				err := cl.Get(ctx, client.ObjectKey{Name: testMapiMachine.Name, Namespace: testMapiMachine.Namespace}, currentMachine)
```

Comment:
How about using [the current function](https://github.com/openshift/cluster-capi-operator/blob/main/e2e/framework/machine.go#L78) to get the machine?